packages/flutter/lib/src/animation/animation_controller.dart

```diff
@@ -969,11 +991,17 @@ class _RepeatingSimulation extends Simulation {
   final double min;
   final double max;
   final bool reverse;
+  final int? repeatCount;
   final _DirectionSetter directionSetter;
 
   final double _periodInSeconds;
   final double _initialT;
 
+  /// If [reverse] is true, the duration of a full cycle is doubled.
+  late final int _effectiveRepeatCount = repeatCount! * (reverse ? 2 : 1);
+
+  late final double _exitTimeInSeconds = (_effectiveRepeatCount * _periodInSeconds) - _initialT;
```

Comment:
If `reverse` is true, we can choose to _not_ double the duration: that way, we can pass an odd number to `count` in order for it to end in the "completed" state.

```suggestion
  late final double _exitTimeInSeconds = (repeatCount! * _periodInSeconds) - _initialT;
```

Reply:
but ideally, people would think when `count` given as 2, the animation shall repeat twice. If we remove this doubling logic, then if user passed 2 then it will only repeat once, isn't it? 

I think we shall keep this doubling logic at controller level to make developer experience more streamline. Please suggest your thoughts on this one.,

Reply:
I think it would be nice to allow the following:

```
forward, reverse, forward, reverse, forward (and then stop)
```

Removing the doubling logic allows the animation to terminate in the "completed" state, and it matches the current behavior shown in a [flutter_animate example](https://youtu.be/lxkFL9disq0?t=544).

Reply:
hmm alright, if you say so then I will update the changes accordingly. Please allow me some time